src/custom_types.ts

```diff
@@ -0,0 +1,116 @@
+const MAX_32_NUM = Math.pow(2,31);
+const MAX_64_NUM = Math.pow(2,63);
+const MAX_U32_NUM = Math.pow(2,32);
+const MAX_U64_NUM = Math.pow(2,64);
+
+export class Int32 {
+	private val: number;
+
+	constructor(n: number) {
+		if(n === null || n === undefined) {
+			throw new Error("Int32 cannot be null or undefined");
+		}
+		if(n >= MAX_32_NUM || n < (-MAX_32_NUM)) {
+			throw new Error("integer out of range (32bit)");
+		} else {
+			this.val = n;
+		}
+	}
+
+	static Parse(s: string) {
+		let num = parseInt(s, 10);
+		return new Int32(num);
+	}
+
+	toNumber(): number {
+		return this.val;
+	}
+}
+
+export function int32(n: number): Int32 {
```

Comment:
named the function in lower case since we cannot have the same identifier for both a class and a function. Same with the rest of the classes and functions, we can change either depending on what we want the type to be named.